animation_nodes/nodes/mesh/set_polygon_material_index.py

```diff
@@ -0,0 +1,40 @@
+import bpy
+from ... data_structures import VirtualLongList
+from ... base_types import AnimationNode, VectorizedSocket
+
+class SetPolygonMaterialIndexNode(bpy.types.Node, AnimationNode):
+    bl_idname = "an_SetPolygonMaterialIndexNode"
+    bl_label = "Set Polygon Material Index"
+    errorHandlingType = "EXCEPTION"
+
+    useMaterialIndexList: VectorizedSocket.newProperty()
+
+    def create(self):
+        self.newInput("Object", "Object", "object", defaultDrawType = "PROPERTY_ONLY")
+        self.newInput(VectorizedSocket("Integer", "useMaterialIndexList",
+                ("Material Index", "materialIndex"), ("Material Indices", "materialIndices")))
+
+        self.newOutput("Object", "Object", "object")
+
+    def execute(self, object, materialIndices):
+        if object is None: return None
+
+        if object.type != "MESH":
+            self.raiseErrorMessage("Object is not a mesh object.")
+
+        if object.mode != "OBJECT":
+            self.raiseErrorMessage("Object is not in object mode.")
+
+        if self.useMaterialIndexList:
+            if min(materialIndices) < 0:
```

Comment:
You can use the `containsValueLowerThan(0)` method here.

Reply:
Done.